### ISSUE TITLE
SteamVR_Behaviour_Pose runtime creation issue

### ISSUE THREAD

**Issue by TonyViT** [ValveSoftware/steamvr_unity_plugin#487]
I discovered an issue when creating runtime SteamVR_Behaviour_Pose objects. If the script gets added as a component at runtime, and then the object containing it gets destroyed, Unity will start throwing NullReferenceExceptions.

I discovered why. When the script gets added, it calls OnEnable, and this triggers this code:
```
if (poseAction != null)
{
       poseAction[inputSource].onUpdate += SteamVR_Behaviour_Pose_OnUpdate;
       poseAction[inputSource].onDeviceConnectedChanged += OnDeviceConnectedChanged;
       poseAction[inputSource].onTrackingChanged += OnTrackingChanged;
       poseAction[inputSource].onChange += SteamVR_Behaviour_Pose_OnChange;
 }
```
The problem is that this gets called with inputSource that is "Any", because the dev can set the inputsource only AFTER the script has been added. When the object gets destroyed, it is called the OnDisabled function, that removes the above events, but now inputSource has the value defined by the user, e.g. "LeftHand", so the events created above are not removed (because the inputSource value is different), and so SteamVR still tries to call the events of a destroyed object. kaboom.

For now, I am using the workaround of disabling the script upon creation, so that to trigger a first OnDisabled with the "Any" source and so clear the events, before setting the new ones. Here you are a sample script that works well:

```
Valve.VR.SteamVR_Behaviour_Pose behaviourPose;
behaviourPose = transform.Find("Hand L").gameObject.AddComponent<Valve.VR.SteamVR_Behaviour_Pose>();
behaviourPose.enabled = false;
behaviourPose.inputSource = Valve.VR.SteamVR_Input_Sources.LeftHand;
behaviourPose.enabled = true;
```

**Comment by zite** [comment]
Thanks for the report! I think what we'll do is switch inputSource to private (but exposed in editor) then add a public function SetInputSource that removes the previous events and adds new events. Sound good?

**Comment by TonyViT** [comment]
Sounds very good to me! Or instead of defining a function, you can offer an InputSource property with related getters and setters.

**Comment by zite** [comment]
I believe this has been fixed since this post was made. If not, please create a new issue with the repro steps and ideally an example project to replicate the issue. Thanks!

**Comment by nikita-mordik** [comment]
@keithbradner I'm experiencing a similar issue: when recreating the PlayerRig, all SteamVR_Behaviour_Pose components on the player return `isActive = false`.
However, when the PlayerRig is created for the first time, everything works as expected.
That happening in build, in Editor evrything work fine.
